src/charm.py

```diff
@@ -157,10 +163,14 @@ def _update_config(self, event=None):
         try:
             if yaml.safe_load(config) != yaml.safe_load(old_config):
                 self._container.push(CONFIG_PATH, config)
+                self._stored.agent_config = config
                 # FIXME: #19
                 # self._reload_config()
                 self._container.restart(self._name)
                 self.unit.status = ActiveStatus()
+        except APIError as e:
+            self.unit.status = WaitingStatus(str(e))
+            event.defer()
         except GrafanaAgentReloadError as e:
             self.unit.status = BlockedStatus(str(e))
```

Comment:
Would it make sense to narrow down the scope of the try, maybe break it into two separate ones?